src/Grpc.Net.Client/Balancer/Internal/BalancerHttpHandler.cs

```diff
@@ -0,0 +1,123 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Shared;
+
+namespace Grpc.Net.Client.Balancer.Internal
+{
+    internal class BalancerHttpHandler : DelegatingHandler
+    {
+        internal const string WaitForReadyKey = "WaitForReady";
+        internal const string SubchannelKey = "Subchannel";
+
+        private readonly ConnectionManager _manager;
+
+        public BalancerHttpHandler(HttpMessageHandler innerHandler, ConnectionManager manager)
```

Comment:
This handler makes sense to me but I'm not the most familiar person with the intricacies of DelegatingHandlers so can we get another set of eyes on this?